core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java

```diff
@@ -272,50 +322,41 @@ public void stopSyncForMount(long mountId) throws InvalidPathException {
     if (mFilterMap.containsKey(mountId)) {
       List<AlluxioURI> toBeDeleted = new ArrayList<>(mFilterMap.get(mountId));
       for (AlluxioURI uri : toBeDeleted) {
-        stopSyncInternal(uri, mountId);
+        stopSyncAndJournal(RpcContext.NOOP, uri);
       }
     }
   }
 
   /**
-   * Perform various checks of stopping a sync point.
-   *
-   * @param syncPoint sync point to stop
-   * @return the path resolution result if successfully passed all checks
-   */
-
-  @Nullable
-  public MountTable.Resolution resolveSyncPoint(AlluxioURI syncPoint) throws InvalidPathException {
-    if (!mSyncPathList.contains(syncPoint)) {
-      LOG.debug("syncPoint not found {}", syncPoint.getPath());
-      return null;
-    }
-    MountTable.Resolution resolution = mMountTable.resolve(syncPoint);
-    return resolution;
-  }
-  /**
-   * stop active sync on a URI.
+   * Stop active sync on a URI and journal the remove entry.
    *
+   * @param rpcContext the master rpc or no-op context
    * @param syncPoint sync point to be stopped
-   * @param mountId mountId
    */
-  public void stopSyncInternal(AlluxioURI syncPoint, long mountId) {
-    try (LockResource r = new LockResource(mSyncManagerLock)) {
+  public void stopSyncAndJournal(RpcContext rpcContext, AlluxioURI syncPoint)
+      throws InvalidPathException {
+    try (LockResource r = new LockResource(mLock)) {
+      MountTable.Resolution resolution = mMountTable.resolve(syncPoint);
+      if (resolution == null) {
+        throw new InvalidPathException(syncPoint + " is not a sync point.");
```

Comment:
NIT: can we say something like "Failed to stop sync point <URI> because it is not mounted."